### PR TITLE
Responsive Facebook posts and videos

### LINKED ISSUE #9: Make Facebook posts and videos "responsive"
It would be nice if Facebook posts and videos were responsive to their parent container. However, the implementation will need to be an extension of #7, as the width for the Facebook post is hard-coded to the iframe src.

From #8

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #9

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@davisshaver @goldenapples Can you give this a sanity check locally, and critique my abstraction? I'm not a huge fan of moving the callback to the Shortcake Bakery class, but don't have any great ideas for alternatives.

**Comment by danielbachhuber** [comment]
jun 25 2015 06 25

File: assets/js/shortcake-bakery.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			var el = $(this),
+				parentWidth = el.parent().width();
+
+			// Inside an iframe
+			if ( window.self !== window.top ) {
+				parentWidth = parent.innerWidth;
+			}
+
+			var trueHeight = el.data('true-height') ? el.data('true-height') : 360;
+			var trueWidth = el.data('true-width') ? el.data('true-width') : 640;
```
Comment (goldenapples) [inline assets/js/shortcake-bakery.js]: This seems like an implementation-specific detail. Should this calculation really go here, or should it be up to the shortcodes to listen to?

Comment (danielbachhuber) [inline assets/js/shortcake-bakery.js]: > Should this calculation really go here, or should it be up to the shortcodes to listen to?

Then each shortcode has to support it? It's an abstraction that's worked for several shortcodes in our theme.

**Comment by danielbachhuber** [comment]
> I'm not a huge fan of moving the callback to the Shortcake Bakery class, but don't have any great ideas for alternatives.

Alternative would need to be an action or a filter. I've filed a core ticket for it https://core.trac.wordpress.org/ticket/32790

**Pull request merged**